filesystems/drs/src/main/scala/cromwell/filesystems/drs/DrsPathBuilder.scala

```diff
@@ -0,0 +1,35 @@
+package cromwell.filesystems.drs
+
+import java.net.URI
+
+import cloud.nio.impl.drs.DrsCloudNioFileSystemProvider
+import com.google.common.net.UrlEscapers
+import cromwell.core.path.{Path, PathBuilder}
+
+import scala.util.{Failure, Try}
+
+
+case class DrsPathBuilder(fileSystemProvider: DrsCloudNioFileSystemProvider) extends PathBuilder {
+
+  private val drsScheme: String = fileSystemProvider.getScheme
+
+  override def name: String = "DRS"
+
+  override def build(pathAsString: String): Try[Path] = {
+    if (pathAsString.startsWith(s"$drsScheme://")) {
+      Try(URI.create(UrlEscapers.urlFragmentEscaper().escape(pathAsString))) flatMap { uri =>
+        if (!Option(uri.getScheme).exists(_.equalsIgnoreCase(fileSystemProvider.getScheme))) {
+          Failure(new IllegalArgumentException(s"$pathAsString does not have a $drsScheme scheme."))
+        } else if (uri.getHost == null && uri.getAuthority == null) {
+          Failure(new IllegalArgumentException(s"$pathAsString does not have a valid host."))
+        } else if (uri.getPath == null || uri.getPath.isEmpty || uri.getPath.equalsIgnoreCase("/")) {
+          Failure(new IllegalArgumentException(s"$pathAsString does not have a valid path. DRS doesn't support a host only path."))
+        } else {
+          Try(DrsPath(fileSystemProvider.getPath(uri), fileSystemProvider.drsPathResolver))
```

Comment:
`java.nio.Path` instances already contain pointers to their `FileSystemProvider`. So I believe the path resolvers should be available via something like `drsPath.getFileSystem.provider.asInstanceOf[DrsCloudNioFileSystemProvider].drsPathResolver`. If so, then a second variable doesn't need to be passed into the `DrsPath` constructor.

Reply:
When we talked about it with Saloni it felt like the `asInstanceOf` was breaking the nio abstraction but if you think it's fine then 👍 for doing that instead of the extra argument

Reply:
Removing the second variable as I can access `drsPathResolver` through `drsPath.getFileSystem.provider.asInstanceOf[DrsCloudNioFileSystemProvider].drsPathResolver`. Thank you! :)